bloggies_frontend/src/RoutedPages/PremiumBlogPage/index.tsx

```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+function PremiumBlogPage() {
```

Comment:
What is the purpose of the PremiumBlogPage component?

Reply:
Thanks for asking! `PremiumBlogPage` will be the component that renders the page for the `paid member-only` posts.

I opt'd for not re-using the `Homepage` component (which is currently being used to render the `free` blogs) because the `PremiumBlogPage` will have additional checks to ensure that the user accessing this route is a `paid member`, as well as it will be used `dispatch` an `action creator` (Redux) to obtain the `paid members-only` posts from the backend. Otherwise, if the user is a `non-paid member`, they should be redirected to a page that notifies them that they do not have permissions to access the `PremiumBlogPage` and will require a `paid membership` to gain access. 
I thought that because those reasons, it would diminish the readability of the code due to the added complexity that would only make sense for the `PremiumBlogPage` to use.

Reply:
I think the filtering out for premium and non-premium blog posts should be handled in the backend server. Then on the frontend we can fetch those posts based on what is premium or not.

Reply:
Yup, that would definitely be the case. The conflict that I had was not the separation of the premium and non-premium blog posts, but how we will be handling the user's paid-membership verification/how we will decide when we will choose to display premium blogs. I thought about it more and this is what I came up with: 

In a reuse case, we might have to manually check the URL bar to verify it is the URL for the premium blog posts section and then make an API call to an endpoint that will verify whether the current user is a `paid member` or not. When it's confirmed the current user is a `verified paid member`, then a fetch will be called for obtaining the premium blog posts. 

If we do decide to separate it out, the difference that I can think of is that, we don't have to make the extra step of checking the URL to match it to the premium blog posts page.

But since we are considering to keep both pages the same layout, I don't think it will hurt to do that extra step. I'll update the name of the `Homepage` component to better represent the component in that case.

Reply:
I think to solve for this we should separate the logic for membership from the front end in terms of retrieving posts. Instead of having two pages for the different type of blog posts, it should be one page since the structure for the post are equivalent. Instead the server should handle what should be sent to the front. This could work by sending the `JWT` for a user to the server. The server with the `JWT` can then check to see if a user is premium or not. Then it should return all posts if the user is `verified paid member` or filter the post if the user is not.